tests/functional/base_test.go

```diff
@@ -262,3 +289,54 @@ func SimulateDaemonsetNumberReadyWithPods(name types.NamespacedName, networkIPs
 
 	logger.Info("Simulated daemonset success", "on", name)
 }
+
+func GetDNSData(name types.NamespacedName) *infranetworkv1.DNSData {
+	dns := &infranetworkv1.DNSData{}
+	Eventually(func(g Gomega) {
+		g.Expect(k8sClient.Get(ctx, name, dns)).Should(Succeed())
+	}).Should(Succeed())
+
+	return dns
+}
+
+func GetDNSDataList(name types.NamespacedName) *infranetworkv1.DNSDataList {
+	dnsList := &infranetworkv1.DNSDataList{}
+	Eventually(func(g Gomega) {
+		g.Expect(k8sClient.List(ctx, dnsList, client.InNamespace(name.Namespace))).Should(Succeed())
+	}).Should(Succeed())
+
+	return dnsList
+}
+
+func GetDNSDataHostnameIP(dnsDataName string, namespace string, dnsHostname string) string {
+	dnsEntry := GetDNSData(types.NamespacedName{Name: dnsDataName, Namespace: namespace})
+	// DNSData contains Hosts (an array), this array will contain
+	// ovsdbserver-sb-0.ns.svc and ovsdbserver-sb.ns.svc, need to get
+	// in which index is located the one which the flavor is currently testing (DNS Cname or DNS)
+	index := -1
+	ip := ""
+	for i, host := range dnsEntry.Spec.Hosts {
+		for _, hostname := range host.Hostnames {
+			if hostname == dnsHostname {
+				index = i
```

Comment:
you can return host.IP here, no? `index` is useless.

Reply:
True, done